src/utils/pyproject.ts

```diff
@@ -40,9 +48,21 @@ function parsePyproject(pyprojectContent: string): string | undefined {
   )
     .flat()
     .filter((item: string | object) => typeof item === "string");
-  return getRuffVersionFromAllDependencies(
+  const version = getRuffVersionFromAllDependencies(
     dependencies.concat(optionalDependencies, devDependencies),
   );
+  if (version) return version;
+
+  // Special handling for Poetry until it supports PEP 735
+  // See: <https://github.com/python-poetry/poetry/issues/9751>
+  const poetryGroups = Object.values(pyproject?.tool?.poetry?.group ?? {});
```

Comment:
```suggestion
  const poetryGroups = Object.values(pyproject?.tool?.poetry?.group || {});
```
nullish coalescing or logical or doesn't(shouldn't) make a big difference here but lets stay consistent.

Reply:
Agreed, force of habit on my part :)